service/core/client/src/main/scala/CoordinatedShutdownSupport.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+// Hack to expose play.api.libs.concurrent.CoordinatedShutdownSupport, which is package private to [play], to be package private to [lagom].
```

Comment:
We had talked before that copying the code was okay since it is just a small portion. But I'm changing my mind and thinking about making this API public in Play. Not a big deal since it is very small.

WDYT?

Reply:
Perhaps the parts of Lagom that need to call this should not be part of Lagom, but part of core Play. In that case, these interface hacks are a good way of finding the places where Lagom has broken encapsulation.

Reply:
Let me elaborate this: the places where Lagom needs to use this is in places where it creates a "client application".

Play has no notion of a "client application" — all Play applications are intrinsically assumed to be server applications and have a lifecycle associated with an HTTP server.

The use case for a Lagom client application is when you want to embed a Lagom service client in a JVM application that is not written using Lagom or Play. More broadly, if you wanted to embed something that uses the Play Application for dependency injection and the module system, but without creating an HTTP server, it would be useful to be able to do this.